app/process/recovery/recovery.go

```
@@ -235,61 +234,61 @@ func (r Recovery) recoverEthereumTXMessage(tm encoding.TopicMessage) error {
+
+	for _, txnMessage := range unprocessedMessages {
+		key := &TransferMessageKey{
+			TransactionId: txnMessage.TransactionId,
+			EthAddress:    txnMessage.EthAddress,
+			Amount:        txnMessage.Amount,
+			Fee:           txnMessage.Fee,
+			GasPriceWei:   txnMessage.GasPriceWei,
+			Erc20Address:  txnMessage.ERC20ContractAddress,
+		}
+		signatureMessagesMap[key] = append(signatureMessagesMap[key], txnMessage.Signature)
+	}
+
+	for txn, txnSignatures := range signatureMessagesMap {
+		hasSubmittedSignature, topicMessage := r.hasSubmittedSignature(txn, txnSignatures)
+
+		if !hasSubmittedSignature {
+			err = r.transfers.PrepareAndSubmitToTopic(topicMessage)
+			if err != nil {
+				r.logger.Errorf("Failed to prepare and submit Transaction ID [%s] to the HCS Topic. Error [%s].", txn.TransactionId, err)
+				return err
+			}
+		}
+	}
 	return nil
 }
 
-func (r Recovery) hasSubmittedSignature(data joined.CTMKey, signatures []string) (bool, *validatorproto.TransferMessage) {
-	//ctm := &validatorproto.TransferMessage{
-	//	TransactionId: data.TransactionId,
-	//	EthAddress:    data.EthAddress,
-	//	Amount:        data.Amount,
-	//	Fee:           data.Fee,
-	//	GasPriceGwei:  data.GasPriceGwei,
-	//}
-	//
-	//signature, err := r.transfersService.ValidateAndSignTxn(ctm)
-	//if err != nil {
-	//	r.logger.Errorf("Failed to Validate and Sign TransactionID [%s]. Error [%s].", data.TransactionId, err)
-	//}
-	//
-	//for _, s := range signatures {
-	//	if signature == s {
-	//		return true, nil
-	//	}
-	//}
-	return false, nil
+func (r Recovery) hasSubmittedSignature(data *TransferMessageKey, signatures []string) (bool, *encoding.TopicMessage) {
+	signature, err := r.transfers.SignAuthorizationMessage(data.TransactionId, data.EthAddress, data.Erc20Address, data.Amount, data.Fee, data.GasPriceWei)
+	if err != nil {
+		r.logger.Errorf("Failed to Validate and Sign TransactionID [%s]. Error [%s].", data.TransactionId, err)
```

We should return the error here and handle it in the iteration above. 
If the `SignAuthorizationMessage` fails, we log the error in the `range` above and state that this `TXID` will not be recovered and you should do a `continue` (skipping this tx recovery due to invalid data as we do in the other places in the recovery